### PR TITLE
Add story kind regex

### LINKED ISSUE #91: Add regex to match the story's kind
What do you guys think about having these two config options: `storyKindRegex` and `storyNameRegex` instead of the current `storyRegex`?

`storyNameRegex` would behave just as the existing `storyRegex`, while `storyKindRegex` would match against the story's kind, giving users more flexibility.

For example, it could be used to solve #84 doing something like this:

```javascript
// SomeComponent.stories.js

// ...
storiesOf('SomeComponent', module)
    .add('foo', () => {
        // ...
    });
```

```javascript
// SomeComponent.test.js

import initStoryshots from 'storyshots';

initStoryshots({
  storyKindRegex: /^SomeComponent$/,
});
```

Which will cause storyshots (or jest actually) to output a `__snapshots__` directory colocated with the component and with just its storyshots.

If you think it is a good idea, I'm happy to submit a PR!

### PR DESCRIPTION
**Pull request by olistic**
This closes #91.

Thanks for considering the PR!

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,10 +52,14 @@ export default function testStorySnapshots (options = {}) {
...
   for (const group of stories) {
+    if (options.storyKindRegex && !group.kind.match(options.storyKindRegex)) {
+      continue
+    }
+
     describe(suit, () => {
       describe(group.kind, () => {
         for (const story of group.stories) {
-          if (options.storyRegex && !story.name.match(options.storyRegex)) {
+          if (options.storyNameRegex && !story.name.match(options.storyNameRegex)) {
```
Comment (thani-sh) [inline src/index.js:65]: Also support `options.storyRegex` so we don't break any storybooks. It's okay to remove it from the Readme file.

**Review by thani-sh — CHANGES_REQUESTED** [review]

**Comment by thani-sh** [comment]
Hi @olistic,

Awesome PR :) Can you make a small change to the code so it'll continue to work with `options.storyRegex` also. Maybe you can add the following line somewhere.

```js
options.storyNameRegex = options.storyNameRegex || options.storyRegex
```

**Comment by olistic** [comment]
Thanks @mnmtanish, the change is done!

**Review by thani-sh — APPROVED** [review]

**Comment by shilman** [comment]
Hey @olistic 

Thank you so much for this PR! We're moving over to a mono-repo, since that makes a lot of sense for a project like this.

We would ❤️ it if you could create a PR here: https://github.com/storybooks/storybook

It looks like the changes have already been approved, so just let me know when it's ready and I'll merge it on the monorepo right away. Thanks and sorry for the hassle!

**Pull request closed without merging**